spec.html

```diff
@@ -21345,6 +21345,7 @@
               </td>
               <td>
                 <p>Return the binding of a name exported by this module. Bindings are represented by a <dfn id="resolvedbinding-record">ResolvedBinding Record</dfn>, of the form {[[Module]]: Module Record, [[BindingName]]: String}. If the name cannot be resolved, return either the module record that is to blame for the resolution failure or, in case there isn't any, return *null* or `"ambiguous`" depending on the nature of the failure.</p>
+                <p>This operation must be idempotent if it completes normally. Each time it is called with a specific _exportName_, _resolveSet_ pair as arguments it must return the same result.</p>
```

Comment:
@domenic @GeorgNeis FYI.

Reply:
Is that true? It seems likely, but I've recently been made aware of how horrifically complicated export resolution is (due to export from and circular dependencies), so I wouldn't want to bet on this at my current level of understanding.

Reply:
It's not necessarily true, but it's the easiest way to enforce that the assertion in step 7 of https://tc39.github.io/ecma262/#sec-module-namespace-exotic-objects-get-p-receiver holds:
> ...
> 6. Let binding be ! m.ResolveExport(P, « »).
> 7. Assert: binding is a ResolvedBinding Record.
> ...

There are other ways to correct this assertion:
- We could make Source Text Module Records the only allowed Module Record type.
- We could disallow non-Source Text Module Records in Source Text Module Records module dependency graphs, and then restrict Module Namespace objects to Source Text Module Records.
- We could store the ResolveExport results in https://tc39.github.io/ecma262/#sec-getmodulenamespace into the Module Namespace object, so that https://tc39.github.io/ecma262/#sec-module-namespace-exotic-objects-get-p-receiver no longer needs to call ResolveExport.
- Or we could just make step 6 in https://tc39.github.io/ecma262/#sec-module-namespace-exotic-objects-get-p-receiver fallible again.

Reply:
@domenic I'm going to pull this, we can select another alternative above later if you like.

Reply:
The added requirement lgtm.

Reply:
There's a similar comment in https://tc39.github.io/ecma262/#sec-module-namespace-exotic-objects-get-p-receiver